### PR TITLE
M3-4201 TableSettings dropdown UI

### PR DESCRIPTION
**Pull request by WilkinsKa1**
## Description

New component to allow user control for table displays (column headers/ordering). This particular PR is strictly handling UI/styling, no functionality has been implemented as of yet (though some of the accessibility work has been handled here). This component expects an aria label, title for the field legend, and a list of options (the button icon is customizable as well, just defaults to the gear icon).

To view, run storybook: http://localhost:6006/?path=/story/table-editor--default

Screen Shot 2020-06-18 at 3 57 54 PM

## Type of Change
- Non breaking change ('update')

### CODE REVIEW

File: packages/manager/src/components/TableEditor.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { storiesOf } from '@storybook/react';
+import * as React from 'react';
+import TableEditor from './TableEditor';
+
+const linodeTableColumns = [
+  { label: 'Status', id: 'linode-status', selected: true },
+  { label: 'CPU', id: 'linode-cpu', selected: false },
+  { label: 'Ram', id: 'linode-ram', selected: false },
```
Comment (DevDW) [inline packages/manager/src/components/TableEditor.stories.tsx:8]: Should be "RAM"

File: packages/manager/src/components/TableEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      </span>
+      <span id="operation" className="visually-hidden">
+        To reorder and customize table display, press Enter key to toggle drag
+        drop mode, use arrow keys to move selected elements.
+      </span>
+
+      <IconButton
+        aria-label={ariaLabel}
+        onClick={handleClick}
+        disableRipple
```
Comment (Jskobos) [inline packages/manager/src/components/TableEditor.tsx:141]: oh thank god

File: packages/manager/src/components/TableEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      >
+        {buttonIcon ? buttonIcon : <Settings />}
+      </IconButton>
+
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        onClose={handleClose}
+        anchorOrigin={{
```
Comment (Jskobos) [inline packages/manager/src/components/TableEditor.tsx:155]: I don't know if we should worry about vertically narrow viewports, but it's possible to break this if the view is too narrow: 

Screen Shot 2020-06-22 at 3 08 03 PM

Comment (Jskobos) [reply packages/manager/src/components/TableEditor.tsx:155]: (the settings icon is covered too, it's not obvious from the shot)

File: packages/manager/src/components/TableEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
+
+  return (
+    <div>
+      <span aria-live="assertive" className="visually-hidden">
+        EXAMPLES ONLY Element 1 grabbed. Current position 1 of 4", "Element 1
+        moved, new positon 2 of 4", and "Element 1 dropped, final position 2 of
```
Comment (johnwcallahan) [inline packages/manager/src/components/TableEditor.tsx:130]: "position", and I'm getting some linter warnings of unescaped characters.
Screen Shot 2020-06-22 at 4 46 11 PM

Comment (johnwcallahan) [reply packages/manager/src/components/TableEditor.tsx:130]: (these examples are cool, though – nice work)

**Comment by WilkinsKa1** [comment]
Thanks all for the feedback! I updated the margins/line height of the list items, in addition to the text updates Dajahi pointed out. Since we don't have the icon exported yet, I'm defaulting to the gear icon, but this component will accept any icon so we can use this in multiple places with differing icon indicators.

@Jskobos for your callouts: space bar should toggle checkbox selection, using a SR enter or spacebar will select. We can add another keypress event for enter explicitly. The overflow issue you called out will need to be addressed separately- if I re-add that we lose the caret arrow :(

**Pull request merged**